clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -174,28 +177,45 @@ public void assignFromUser(Set<TopicPartition> partitions) {
     }
 
     /**
-     * Change the assignment to the specified partitions returned from the coordinator,
-     * note this is different from {@link #assignFromUser(Set)} which directly set the assignment from user inputs
+     * Change the assignment to the specified partitions returned from the coordinator, note this is
+     * different from {@link #assignFromUser(Set)} which directly set the assignment from user inputs.
+     *
+     * @return true if assignments matches subscription, otherwise false
      */
-    public void assignFromSubscribed(Collection<TopicPartition> assignments) {
+    public boolean assignFromSubscribed(Collection<TopicPartition> assignments) {
```

Comment:
It may be helpful to have some focused unit tests in `SubscriptionStateTest`.

Reply:
Good catch. We have many tests in `SubscriptionStateTest`. I updated all of them to assert the correct value is returned.